process/block/preprocess/transactions.go

```diff
@@ -731,11 +731,23 @@ func (txs *transactions) getAllTxsFromMiniBlock(
 	return txsSlice, txHashes, nil
 }
 
+func (txs *transactions) getRemainingGasPerBlock() uint64 {
+	gasConsumed := txs.gasHandler.TotalGasConsumed()
+	maxGasPerBlock := txs.economicsFee.MaxGasLimitPerBlock(txs.shardCoordinator.SelfId())
```

Comment:
Use MaxSafeGasLimitPerBlock

Reply:
MaxGasLimitPerBlockForSafeCrossShard

Reply:
Hm.. could remain as it is if we see this from the sender point of view as gas consumed in self shard